src/include/add-on/add-on-workflow.rb

```diff
@@ -354,26 +354,30 @@ def InstallProduct
         return :next
       end
 
-      all_products = Pkg.ResolvableProperties("", :product, "")
+      all_products = Y2Packager::Resolvable.find(kind: :product)
       all_products.each do |product|
         # Product doesn't match the new source ID
-        next unless @added_repos.include?(product["source"])
+        next unless @added_repos.include?(product.source)
         # Product is not available (either `installed or `selected or ...)
-        if product["status"] != :available
-          log.info("Skipping product #{product["name"].inspect} with status " \
-            "#{product["status"].inspect}")
+        if product.status != :available
+          log.info("Skipping product #{product.name} with status " \
+            "#{product.status}")
           next
         end
 
-        success = Pkg.ResolvableInstall(product["name"], :product)
+        success = Pkg.ResolvableInstall(product.name, :product)
 
-        log.info("Selecting product '#{product["name"]}' for installation -> #{success}")
+        log.info("Selecting product '#{product.name}' for installation -> #{success}")
       end
 
       :next
     end
 
     def ProductSelect
+      # Do not use Y2Packager::Resolvable.find(kind: :product) because the
+      # returned Resolvables have not the variable "media".
```

Comment:
I am not sure here, if "media" is a valid call at all. Lada, do you know more about it ?